frontend/packages/local-storage-operator-plugin/src/components/local-volume-set/create-local-volume-set.tsx

```diff
@@ -1,67 +1,65 @@
+import { Form, TextVariants } from '@patternfly/react-core';
+import { resourcePathFromModel, BreadCrumbs } from '@console/internal/components/utils';
 import { history } from '@console/internal/components/utils/router';
 import { k8sCreate, NodeKind, referenceForModel } from '@console/internal/module/k8s';
 import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
-import { getName } from '@console/shared';
 import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager';
 import { LocalVolumeSetModel } from '../../models';
-import { LocalVolumeSetHeader, LocalVolumeSetInner } from './local-volume-set-inner';
 import { reducer, initialState } from './state';
-import { nodeResource } from '../../constants/resources';
-import { hasNoTaints, createMapForHostNames } from '../../utils';
-import { getLocalVolumeSetRequestData } from './local-volume-set-request-data';
-
+import { nodeResource } from '../../resources';
+import { hasNoTaints, getNodesByHostNameLabel } from '../../utils';
+import { getLocalVolumeSetRequestData } from './request';
+import { LocalVolumeSetBody } from './body';
+import { LocalVolumeSetHeader } from './header';
+import { FormFooter } from '../form-footer';
 import './create-local-volume-set.scss';
 
-const CreateLocalVolumeSet: React.FC = withHandlePromise<
-  CreateLocalVolumeSetProps & HandlePromiseProps
->((props) => {
-  const { t } = useTranslation();
+const CreateLocalVolumeSet: React.FC<CreateLocalVolumeSetProps> = ({ match }) => {
+  const { appName, ns } = match.params;
+  const resourcePath = resourcePathFromModel(ClusterServiceVersionModel, appName, ns);
 
-  const { match, handlePromise, inProgress, errorMessage } = props;
+  const { t } = useTranslation();
   const [state, dispatch] = React.useReducer(reducer, initialState);
-  const [nodeData, nodeLoaded, nodeLoadError] = useK8sWatchResource<NodeKind[]>(nodeResource);
-
-  const { appName, ns } = match.params;
+  const [nodesData, nodesLoaded, nodesLoadError] = useK8sWatchResource<NodeKind[]>(nodeResource);
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMessage, setError] = React.useState('');
 
   React.useEffect(() => {
-    if ((nodeLoadError || nodeData.length === 0) && nodeLoaded) {
-      dispatch({ type: 'setNodeNamesForLVS', value: [] });
-    } else if (nodeLoaded) {
-      const allNodeNames = nodeData.filter(hasNoTaints).map(getName);
-      const hostNames = createMapForHostNames(nodeData);
-      dispatch({ type: 'setNodeNamesForLVS', value: allNodeNames });
-      dispatch({ type: 'setHostNamesMapForLVS', value: hostNames });
+    if (nodesLoaded && !nodesLoadError && nodesData?.length !== 0) {
+      const filteredNodes: NodeKind[] = nodesData.filter(hasNoTaints);
+      dispatch({ type: 'setLvsAllNodes', value: filteredNodes });
     }
-  }, [nodeData, nodeLoaded, nodeLoadError]);
+  }, [nodesData, nodesLoadError, nodesLoaded]);
```

Comment:
Wouldn't nodesData retrigger this everytime? Perhaps we should memoize it?

Reply:
It wont be much fruitful but possibly add more calculations.
The nodes data carries a couple of more fields like images, sizeinBytes, conditions, which actually change frequently.
If we want to memoize then we need to maintain a particular memoixed functions which checks set of values - memory, name, zones, role which are specific to ocs.

Reply:
Use `deepCompareMemoize` and use the memiozed value in the callback. No unecessary triggers.

Reply:
I think you missed my point. Since the frequency of getting new data is more  here `deepCompareMemoize` will most of the times be false.